framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/EntityHelper.cs

```diff
@@ -109,5 +112,26 @@ public static Type FindPrimaryKeyType([NotNull] Type entityType)
 
             idProperty.SetValue(entity, idFactory());
         }
+
+        public static object GetEntityId(object entity)
+        {
+            if (!IsEntity(entity.GetType()))
+            {
+                throw new AbpException(entity.GetType() + " is not an Entity !");
+            }
+
+            return ReflectionHelper.GetValueByPath(entity, entity.GetType(), "Id");
```

Comment:
hi
`entity` that implement the `IEntity `interface do not have an `Id` property.

Reply:
Hello maliming,
So you mean if an entity is inherited from `Entity` (or `IEntity`) it won't pass this check which will be causing problems. Do I get it right?

Reply:
The Entity will pass the check, but it does not have an `Id `property. Only entities that inherit the `IEntity<IdKeyType>` interface will have an `Id` property.

Reply:
Ok got it now..
So it should be something like;
```
if (!ReflectionHelper.IsAssignableToGenericType(entity.GetType(), typeof(IEntity<>)))
{
     throw new AbpException(entity.GetType() + " is not an Entity with an Id property !");
}
```
Seems good to go by this?

Reply:
I think the `GetKeys `method of entity can be used. This way we can delete the entities that implement `IEntity `and `IEntity<T>`.

https://github.com/abpframework/abp/blob/654eeede81a733382066bfd00ab19b6b65ceec0e/framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/Entity.cs#L18

Reply:
I am not sure if it is viable. If an entity has no `Id` property, it can be many-to-many entity without Id property (juntion table) which we wouldn't want to hard delete them all right?

I may have lack of knowledge about use cases of `Entity` and `IEntity` though.

Reply:
There are some documents we can refer to.
https://docs.abp.io/en/abp/latest/Entities#entities-with-composite-keys